plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -272,8 +277,43 @@ public void dropSchema(ConnectorSession session, String schemaName)
     @Override
     public void renameSchema(ConnectorSession session, String schemaName, String newSchemaName)
     {
-        // TODO: https://github.com/trinodb/trino/issues/10558
-        throw new TrinoException(NOT_SUPPORTED, "This connector does not support renaming schemas");
+        String schemaEngineType = getSchemaEngine(session, schemaName).orElseThrow(() -> new TrinoException(SCHEMA_NOT_FOUND, format("Schema '%s' does not exist", schemaName)));
+        ClickHouseDatabaseEngineType schemaEngine = fromEngineType(schemaEngineType);
+
+        if (schemaEngine == null) {
+            throw new TrinoException(NOT_SUPPORTED, format("Database engine '%s' is not supported", schemaEngineType));
+        }
+        if (schemaEngine != ClickHouseDatabaseEngineType.ATOMIC) {
+            throw new TrinoException(NOT_SUPPORTED, "This connector does not support renaming schemas");
+        }
+
+        String sql = format(
+                "RENAME DATABASE %s TO %s",
+                quoted(schemaName),
+                quoted(newSchemaName));
+        execute(session, sql);
+    }
+
+    private Optional<String> getSchemaEngine(ConnectorSession session, String schemaName)
+    {
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            String sql = format(
+                    "SELECT engine FROM system.databases WHERE name = '%s'",
+                    schemaName);
+            log.debug("Execute: %s", sql);
+
+            try (Statement statement = connection.createStatement()) {
```

Comment:
Use PreparedStatement. You can refer to `PostgreSqlClient.getArrayColumnDimensions` as an example.